src/libs/ajax/loaded-data/useLoadedData.ts

```diff
@@ -50,11 +64,15 @@ export const useLoadedData = <T>(): UseLoadedDataResult<T> => {
       const error = isFetchResponse(err) ?
         Error(await err.text()) :
         err
-      setLoadedData({
+      const errorResult: ErrorState<T, unknown> = {
```

Comment:
It's a shame that the type can't be inferred here. In Java there's an idiom of static factories that can be used to infer a generic's type arguments from the factory parameters which avoids this kind of redundancy. But I don't know if there's an idiomatic way to do this in typescript.

Reply:
I think we want to be careful with "over-inferring" in Typescript.  I greatly value clarity of developer intent over "brevity".  The starting point here is JavaScript, a full-on dynamic language, as opposed to Java that is statically typed from the get-go.  We want to bias on the side of leveraging Typescript to express developer intent and adding an extra layer of type safety for later code edits.